scripts/tests/test_zkai_native_attention_mlp_adapter_opening_geometry_budget_gate.py

```
@@ -0,0 +1,136 @@
+        rows = context["accounting_sources"]["rmsnorm_input_fused"]["payload"]["rows"]
+        rows[0]["local_binary_accounting"]["grouped_reconstruction"]["unexpected"] = 1
+
+        with self.assertRaisesRegex(gate.AdapterOpeningGeometryBudgetError, "key drift"):
+            gate.build_payload(context)
+
+    def test_cross_source_accounting_drift_is_rejected(self) -> None:
+        context = gate.build_context()
+        rows = context["accounting_sources"]["preprocessed_output_anchor"]["payload"]["rows"]
+        rows[0]["local_binary_accounting"]["component_sum_bytes"] += 1
+
+        with self.assertRaisesRegex(gate.AdapterOpeningGeometryBudgetError, "cross-source accounting drift"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)